crates/router/src/services/authentication.rs

```diff
@@ -852,6 +903,7 @@ where
         let auth = AuthenticationData {
             merchant_account: merchant,
             key_store,
+            profile_id: payload.profile_id,
```

Comment:
Are we not supposed to validate this profile_id?

Reply:
what kind of validation?

Reply:
check if the profile belongs to the given merchant.